### ISSUE TITLE
Can't reproduce results

### ISSUE THREAD

**Issue by m3at** [ByteDance-Seed/Seed-X-7B#8]
Hi, thank you for the release, the model looks promising.

Unfortunately, so far I can't reproduce the results. The [RM_demo.py](https://github.com/ByteDance-Seed/Seed-X-7B/blob/main/RM_demo.py) example have a few issues, but even after fixing them I don't get the expected scores:

```python
prompt = "Translate the following English sentence into Chinese:\nMay the force be with you <zh>"
candidate = ["愿原力与你同在","愿力量与你同在"]
# Expected: [1.46875, -0.376953125]
# Seen: [ 0.3320, -1.7891]
```

<details>

<summary>Fixed code snippet</summary>

```python
import torch
import torch.nn as nn
import transformers
from transformers import AutoTokenizer, MistralForCausalLM
from huggingface_hub import hf_hub_download, try_to_load_from_cache
from safetensors.torch import load_file


class RewardModel:
    def __init__(self, device = torch.device("cuda"), dtype = torch.bfloat16) -> None:
        seed_x_repo = "ByteDance-Seed/Seed-X-RM-7B"
        self.device = device
        self.torch_dtype = dtype

        self.tokenizer = AutoTokenizer.from_pretrained(seed_x_repo)
        self.tokenizer.pad_token = self.tokenizer.eos_token
        self.tokenizer.padding_side = "right"
        
        self.model = MistralForCausalLM.from_pretrained(
            seed_x_repo,
            device_map=device,
            torch_dtype=self.torch_dtype,
            # attn_implementation="flash_attention_2",
            # Ignore mismatched layer
            ignore_mismatched_sizes=True,
        )
        # config.hidden_size=4096
        self.model.lm_head = nn.Linear(4096, 1, bias=False, device=self.device, dtype=self.torch_dtype)

        # Load again
        _path_weights = try_to_load_from_cache(seed_x_repo, "model.safetensors")
        assert _path_weights is not None
        self.model.load_state_dict(load_file(_path_weights), strict=True)

        self.model.to(self.torch_dtype)
        self.model.eval()

    @torch.inference_mode()
    def score(self, prompt, candidates) -> list[float]:
        # Tokenize and pad
        eos = self.tokenizer.eos_token
        b = [prompt + x + eos for x in candidates]
        x = self.tokenizer(b, padding=True, return_tensors="pt").to(self.device)

        # Get all logits
        logits = self.model(x["input_ids"]).logits

        # Get the logits at the first EOS location
        mask = x["attention_mask"]
        cols = torch.arange(mask.size(1), device=mask.device).expand_as(mask)
        last_idx = torch.where(mask.bool(), cols, -1).amax(dim=1).unsqueeze(1)
        logits_eos = logits.squeeze(2).gather(1, last_idx).squeeze(1)

        return logits_eos

remo = RewardModel(device=torch.device("cuda"))

prompt = "Translate the following English sentence into Chinese:\nMay the force be with you <zh>"
candidate = ["愿原力与你同在","愿力量与你同在"]
# Expected: [1.46875, -0.376953125]
# Seen: [ 0.3320, -1.7891]
scores = remo.score(prompt, candidate)
print(scores)
```

</details>

Testing on some samples in other languages, the performance seems low as well:

```python
prompt = "Translate the following English sentence into French:\nI like apples <fr>"
candidate = [
    "J'aime les pommes",
    "j'aime les pommes",
    "j'aime la pomme",
    "n'importe quoi",
    "une autre phrase",
]
scores = remo.score(prompt, candidate)
print("\n".join(f"{s:<5.2f} {c}" for s, c in zip(scores.cpu(), candidate)))
# -0.73 J'aime les pommes
# -2.17 j'aime les pommes
# -5.16 j'aime la pomme
# -3.56 n'importe quoi
# -4.25 une autre phrase
```

**Comment by m3at** [comment]
```diff
        # Tokenize and pad
        eos = self.tokenizer.eos_token
+       bos = self.tokenizer.bos_token
-       b = [prompt + x + eos for x in candidates]
+       b = [prompt + bos + x + eos for x in candidates]
```

Mainly fixed with the change above, the output is now close enough (`[ 1.5156, -0.3828]`).  
[Thanks to](https://huggingface.co/ByteDance-Seed/Seed-X-RM-7B/discussions/2) @tseyde !

Note that sadly the scoring on other languages does not seem ideal:

```
>>> I like apples <fr>
0.49  J'aime les pommes
-2.39 j'aime les pommes
-5.66 j'aime la pomme
-3.72 n'importe quoi
-3.88 une autre phrase
```
